rust/operator-binary/src/kerberos.rs

```diff
@@ -52,29 +55,14 @@ impl HdfsSiteConfigBuilder {
 }
 
 impl CoreSiteConfigBuilder {
-    pub fn security_config(
-        &mut self,
-        hdfs: &HdfsCluster,
-        hdfs_name: &str,
-        hdfs_namespace: &str,
-    ) -> &mut Self {
+    pub fn security_config(&mut self, hdfs: &HdfsCluster) -> Result<&mut Self, Error> {
```

Comment:
**nit:** The former decision to pass validated parameters, was not so bad.

Reply:
We  I basically created this PR because of https://github.com/stackabletech/hdfs-operator/pull/424#discussion_r1396878200.
I think I'm with @maltesander  here, as I needed to pass them around everywhere...